### ISSUE TITLE
[oracle] 'Found unparsable section' with SET

### ISSUE THREAD

**Issue by osarier** [sqlfluff/sqlfluff#8073]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Linting the example below leads to a 'Found unparsable section' error.

### Expected Behaviour

The example should be parsed correctly.
See https://docs.oracle.com/en/database/oracle/oracle-database/26/sqpug/SET.html
and https://docs.oracle.com/en/database/oracle/oracle-database/26/sqpug/SET-system-variable-summary.html

### Observed Behaviour

Linting the example below gives the following error:
`Line 1, Position 1: Found unparsable section: 'set scan off'`

### How to reproduce

```sql
set scan off
``` 

### Dialect

oracle

### Version

4.2.1
python 3.11.3

### Configuration

[sqlfluff]
dialect = oracle

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
